File: src/test/java/io/vproxy/vfx/intro/IntroTheme.java
package io.vproxy.vfx.intro;

import io.vproxy.vfx.manager.font.FontManager;
import io.vproxy.vfx.manager.font.FontProvider;
import io.vproxy.vfx.manager.font.FontSettings;
import io.vproxy.vfx.theme.impl.DarkTheme;
import io.vproxy.vfx.theme.impl.DarkThemeFontProvider;

public class IntroTheme extends DarkTheme {
    @Override
    public FontProvider fontProvider() {
        return new IntroFontProvider();
    }

    public static class IntroFontProvider extends DarkThemeFontProvider {
        @Override
        protected void defaultFont(FontSettings settings) {
            super.defaultFont(settings);
            settings.setFamily(FontManager.FONT_NAME_JetBrainsMono);
        }
    }
}

File: src-ts/src/intro/IntroTheme.ts
import { FontManager } from '../manager/font/FontManager.js';
import type { FontProvider } from '../manager/font/FontProvider.js';
import type { FontSettings } from '../manager/font/FontSettings.js';
import { DarkTheme } from '../theme/impl/DarkTheme.js';
import { DarkThemeFontProvider } from '../theme/impl/DarkThemeFontProvider.js';

export class IntroTheme extends DarkTheme {
  override fontProvider(): FontProvider {
    return new IntroFontProvider();
  }
}

export class IntroFontProvider extends DarkThemeFontProvider {
  protected override defaultFont(settings: FontSettings): void {
    super.defaultFont(settings);
    settings.setFamily(FontManager.FONT_NAME_JetBrainsMono);
  }
}
